### DISCUSSION TITLE
v4.0.2

### DISCUSSION THREAD

**Discussion by regalialong** [AdGoBye/AdGoBye#98] [Announcements]
I usually batch bugfixes instead of doing releases for single fixes, but this one is critical.

![Cat sitting in front of frying pan with flames inside with the caption "Who let bro cook 😭"](https://github.com/user-attachments/assets/9a9e4168-435d-43d4-aad8-2dfb3486d40d)

This is a fix for #90 per #97. v4.0.0 and v4.0.1 are affected by this. 

While adding a patch that fixes database locking when operating on a lot of files concurrently, I accidentally introduced a logic error where we attempt to queue all Content unconditionally to be added to the database.

What should have happened is that we either removed it or checked if they were valid to be added, but that happened at a later stage which made me not catch this (and I retrospectively realized that we couldn't remove items from the queue due to implementation details).

This sometimes resolved itself with the magic of Entity Framework, but this fails when we try to add Imposters (who may share the same ID with existing Avatars).

</br>
tl;dr AGB is adding imposters and duplicates to the Index when it shouldn't which might lead to exceptions, use this release instead thanks

---
Extract the app and run the binary in your terminal. If it can determine your game's working folder, it will generate an index and store it in the same folder as the app.

AdGoBye ships no blocklists by default but we maintain our own blocklists at https://github.com/AdGoBye/AdGoBye-Blocklists, you can add them by adding the URLs in your `appsettings.json` as follows: (now includes `AGBSupporters` for Supporter Boards / Images!)
```json
"BlocklistURLs": [
"https://raw.githubusercontent.com/AdGoBye/AdGoBye-Blocklists/main/AGBBase.toml",
"https://raw.githubusercontent.com/AdGoBye/AdGoBye-Blocklists/main/AGBCommunity.toml",
"https://raw.githubusercontent.com/AdGoBye/AdGoBye-Blocklists/main/AGBUpsell.toml",
"https://raw.githubusercontent.com/AdGoBye/AdGoBye-Blocklists/main/AGBSupporters.toml"
]
```
If you have any problems or suggestions, feel free to tell us in the [Issues](https://github.com/AdGoBye/AdGoBye/issues) or [Discussions](https://github.com/AdGoBye/AdGoBye/discussions).

If you dread our release speed, we generate artifacts for every commit at https://github.com/AdGoBye/AdGoBye/actions
